omegaconf/basecontainer.py

```diff
@@ -490,6 +492,8 @@ def wrap(key: Any, val: Any) -> Node:
                 else:
                     is_optional = target._is_optional()
                     ref_type = target._metadata.ref_type
+            if input_config:
+                val = val._value()
             return _maybe_wrap(
                 ref_type=ref_type,
                 key=key,
```

Comment:
If input is a config why do you unwrap and wrap it?

Reply:
That's because maybe wrap doesn't really wrap nodes and we need to check if the container value(input_config) has the correct types given a ref_type.  

If we remove this the following test case fails(merging complex dicts):
```
(
   "ComplexDict", # Dict[str, Dict[str, List[int]]]
   {
        "dict": {"foo2": {"var2": ["invalid", 4]}},
   },
),
# merge with ComplexDict
```